olap/src/aggregator/ceramic_patch.rs

```diff
@@ -60,19 +61,32 @@ impl WindowUDFImpl for CeramicPatch {
         Ok(Box::new(CeramicPatchEvaluator))
     }
 }
+// Small wrapper container around the data/state fields to hold
+// other mutable metadata for the event.
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct DataContainer<D> {
+    metadata: BTreeMap<String, serde_json::Value>,
+    data: D,
```

Comment:
if this is specifically used by MIDs, should we change `<D>` to explicitly be `json_patch::PatchOperation` instead?

Reply:
No, for init events its not patch operation but the initial document. Whether we treat it as a patch depends on where in the stream the event lives. See the `apply_patch` function just below this one, we use the DataContainer struct twice with two different types for D.